### DISCUSSION TITLE
Debugging is way too difficult

### DISCUSSION THREAD

**Discussion by elsalant** [fybrik/fybrik#552] [Ideas]
Is there some way to percolate up to the M4DApplication the reason for failure?  Here is an example:
> salant@sgx1-lnx:~/hello-world-read-module$ kubectl get M4DApplication
NAME                           AGE
hello-world-read-module-test   23m.   

-Looks fine to me...
-Digging a bit deeper:

>kubectl describe M4DApplication hello-world-read-module-test -n default
Status:
  Conditions:
    Status:  True
    Type:    Failure
    Status:  False
    Type:    Error
  Generated:
    Kind:               Plotter
    Name:               hello-world-read-module-test-default
    Namespace:          m4d-system

- Nothing information here.  No idea what could be causing the error.

- It is only when one describes the blueprint pod that one sees:
- 
Events:
  Type     Reason     Age                   From               Message
  ----     ------     ----                  ----               -------
  Normal   Scheduled  14m                   default-scheduler  Successfully assigned m4d-blueprints/hello-world-read-module-test-default-hello-worl-c2e64-75c9x2s65 to kind-control-plane
  Normal   Pulling    13m (x4 over 14m)     kubelet            Pulling image "ghcr.io/elsalant/hello-world-read-module:v1"
  Warning  Failed     13m (x4 over 14m)     kubelet            Failed to pull image "ghcr.io/elsalant/hello-world-read-module:v1": rpc error: code = Unknown desc = failed to pull and unpack image "ghcr.io/elsalant/hello-world-read-module:v1": failed to resolve reference "ghcr.io/elsalant/hello-world-read-module:v1": failed to authorize: failed to fetch anonymous token: unexpected status: 401 Unauthorized
  Warning  Failed     13m (x4 over 14m)     kubelet            Error: ErrImagePull
  Warning  Failed     12m (x6 over 14m)     kubelet            Error: ImagePullBackOff
  Normal   BackOff    4m20s (x43 over 14m)  kubelet            Back-off pulling image "ghcr.io/elsalant/hello-world-read-module:v1"

- Could this error ("Warning  Failed     13m (x4 over 14m)     kubelet            Failed to pull image "ghcr.io/elsalant/hello-world-read-module:v1": rpc error: code = Unknown desc = failed to pull and unpack image "ghcr.io/elsalant/hello-world-read-module:v1": failed to resolve reference "ghcr.io/elsalant/hello-world-read-module:v1": failed to authorize: failed to fetch anonymous token: unexpected status: 401 Unauthorized") be propagated up to the Application status?

**Comment by simanadler** [comment]
@froesef Your thoughts?
